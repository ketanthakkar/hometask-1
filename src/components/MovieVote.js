import React from 'react';
import '../styleModules/MovieVote.css';

const MovieVote = ({vote}) => <div className = 'movieVote'>{vote}</div>;

export default MovieVote;
