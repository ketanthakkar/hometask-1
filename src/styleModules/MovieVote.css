:root {
    --vote-color: green;
}
.movieVote {
    width:40px;
    height: 40px;
    color:var(--vote-color);
    font-size: 1.2em;
    border:solid 1px var(--vote-color);
    border-radius: 100%;
    display:flex;
    justify-content: center;
    align-items: center;
}