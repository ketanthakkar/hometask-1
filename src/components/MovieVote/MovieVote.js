import React from 'react';
import './MovieVote.css';

const MovieVote = ({vote}) => <div className = 'movieVote'>{vote}</div>;

export default MovieVote;
