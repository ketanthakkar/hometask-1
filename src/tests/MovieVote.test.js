import React from 'react';
import { shallow } from 'enzyme';
import MovieVote from '../components/MovieVote';

describe('MovieVote', () => {
  it('shoud render MovieVote component', () => {
    const render = shallow(
      <MovieVote vote = 'vote' />
    );
    expect(render.find('div').hasClass('movieVote')).toBe(true);
    expect(render).toMatchSnapshot();
  })

  it('should not modify props', () => {
    const vote = 'vote';
    const render = shallow(
      <MovieVote vote = 'vote' />
    );
    expect(vote).toEqual('vote');
    expect(render).toMatchSnapshot();
  })
})