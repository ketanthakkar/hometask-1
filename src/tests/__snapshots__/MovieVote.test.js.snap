// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieVote shoud render MovieVote component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieVote
    vote="vote"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "vote",
      "className": "movieVote",
    },
    "ref": null,
    "rendered": "vote",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "vote",
        "className": "movieVote",
      },
      "ref": null,
      "rendered": "vote",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MovieVote should not modify props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieVote
    vote="vote"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "vote",
      "className": "movieVote",
    },
    "ref": null,
    "rendered": "vote",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "vote",
        "className": "movieVote",
      },
      "ref": null,
      "rendered": "vote",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
